src/System.Threading.Tasks/tests/CancellationTokenTests.netcoreapp.cs

```diff
@@ -26,5 +21,43 @@ public static void CancellationTokenRegistration_Token_MatchesExpectedValue()
                 Assert.Equal(cts.Token, ctr.Token);
             }
         }
+
+        [Fact]
+        public static void CancellationTokenRegistration_UnregisterOnDefaultIsNop()
+        {
+            Assert.False(default(CancellationTokenRegistration).Unregister());
+        }
+
+        [Fact]
+        public static void CancellationTokenRegistration_UnregisterRemovesDelegate()
+        {
+            var cts = new CancellationTokenSource();
+            bool invoked = false;
+            var ctr = cts.Token.Register(() => invoked = true);
+            Assert.True(ctr.Unregister());
+            Assert.False(ctr.Unregister());
+            cts.Cancel();
+            Assert.False(invoked);
+        }
+
+        [Fact]
+        public static void CancellationTokenRegistration_UnregisterDoesntWaitForCallbackToComplete()
+        {
+            using (var barrier = new Barrier(2))
+            {
+                var cts = new CancellationTokenSource();
+                var ctr = cts.Token.Register(() =>
+                {
+                    barrier.SignalAndWait();
+                    barrier.SignalAndWait();
+                });
+
+                Task.Run(() => cts.Cancel());
+
+                barrier.SignalAndWait();
```

Comment:
>barrier.SignalAndWait(); [](start = 16, length = 24)

if you add some little comment here describing the expected behavior would be nice for anyone reading this code.